plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -970,6 +970,10 @@ public Optional<PreparedQuery> implementJoin(
             Map<JdbcColumnHandle, String> leftAssignments,
             JoinStatistics statistics)
     {
+        if (joinType == JoinType.FULL_OUTER) {
+            // FULL JOIN is only supported with merge-joinable or hash-joinable join conditions
```

Comment:
this suggests in future we should look into making available the `joinType` to `isSupportedJoinCondition` so that we can add logic to check that for `FULL_OUTER` the `joinCondition` is as Postgres supports and allow pushdown in those cases.

(No change requested but maybe we create an issue about possible future enhancement).

Reply:
Issue added https://github.com/trinodb/trino/issues/14929